spatialos-sdk/src/worker/parameters.rs

```diff
@@ -430,3 +408,112 @@ impl ThreadAffinityParameters {
         }
     }
 }
+
+/// Helper struct for converting `ConnectionParameters` into `Worker_ConnectionParameters`.
+pub(crate) struct IntermediateConnectionParameters<'a> {
+    params: &'a ConnectionParameters,
+    protocol: IntermediateProtocolType,
+}
+
+impl<'a> IntermediateConnectionParameters<'a> {
+    pub(crate) fn as_raw(&self) -> Worker_ConnectionParameters {
+        let partial_network_params = Worker_NetworkParameters {
+            use_external_ip: self.params.network.use_external_ip as u8,
+            connection_timeout_millis: self.params.network.connection_timeout_millis,
+            default_command_timeout_millis: self.params.network.default_command_timeout_millis,
+
+            ..Worker_NetworkParameters::default()
+        };
+
+        let network = match &self.protocol {
+            &IntermediateProtocolType::Tcp(tcp) => {
+                Worker_NetworkParameters {
+                    connection_type:
+                        Worker_NetworkConnectionType_WORKER_NETWORK_CONNECTION_TYPE_TCP as u8,
+                    tcp,
+
+                    ..partial_network_params
+                }
+            }
+
+            IntermediateProtocolType::Udp {
```

Comment:
Why is there no & before `IntermediateProtocolType`?

Reply:
In the above match arm, having the explicit `&` causes the `tcp` variable to be a copy of the value in `self.protocol`, wheras not having the `&` causes it to have the default behavior which is to borrow the data (which we want for the UDP case since we want pointers back into `self.protocol`.

I realize that it's pretty confusing in this case. Instead I can remove the `&` above and change it do `tcp: *tcp` to make the copy more explicit.

Reply:
Made the change, hopefully that makes things a bit less confusing 🙂